### PR TITLE
Delete datasource command

### PR DESCRIPTION
**Pull request by CaprinaeWT**
## Pull request type

Jira Link: [INT-](url)

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## How to test this PR

The delete-datasources command enables you to remove all datasources within the designated space. By utilizing the --by-slug option, you can filter the datasources based on their slugs, selectively deleting specific datasources. Similarly, the --by-name option functions in the same way, allowing you to filter and delete datasources based on their names.

`storyblok delete-datasources --space-id <SPACE_ID> # Will delete all datasources`
`storyblok delete-datasources --space-id <SPACE_ID> --by-slug global-translations   # Will only delete datasources where the slug starts with global-translations`

1. Verify the command availability: Confirm that the delete-datasources command with the --by-slug and --by-name options is available in the CLI tool or interface you are using.
2. Test scenario 1 - Delete all datasources: Execute the delete-datasources command without any additional options to delete all datasources within the specified space. Verify that all datasources are indeed deleted and the space is empty.
3. Test scenario 2 - Delete by slug: Use the --by-slug option to specify a particular slug value and execute the delete-datasources command. Confirm that only the datasources matching the provided slug are deleted, while other datasources remain unaffected.
4. Test scenario 3 - Delete by name: Utilize the --by-name option to specify a specific name and run the delete-datasources command. Validate that only the datasources with the specified name are deleted, while other datasources persist. 
5. Perform edge cases: Execute the commands with various edge cases, such as non-existent slugs or names, multiple matching datasources, and combinations of options, to ensure the behavior is as expected in different scenarios.

## What is the new behavior?

1. `delete-datasources` command: The command should allow you to delete all datasources within the specified space.
2. `--by-slug` option: When using the `--by-slug` option, you should be able to filter the datasources based on their slugs. Only the datasources matching the provided slug value should be deleted, while others remain unaffected.
3. `--by-name` option: With the `--by-name` option, you should be able to filter the datasources by their names. Only the datasources with the specified name should be deleted, while the rest should remain untouched.

In summary, the expected behavior is that the `delete-datasources` command, along with the `--by-slug` and `--by-name` options, should allow for selective deletion of datasources based on slug and/or name filters, while preserving other datasources in the specified space.

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -505,6 +505,30 @@ program
     }
   })
 
+  // pull-components
```
Comment (ademarCardoso) [inline src/cli.js]: You can remove this comment or add `// delete-datasources`

File: src/tasks/delete-datasources.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const chalk = require('chalk')
+
+/**
+ * @method deleteDatasources
+ * @param  {Object} api
+ * @param  {Object} options { fileName: string, separateFiles: Boolean, path: String }
+ * @return {Promise<Object>}
+ */
+const pullComponents = async (api, options) => {
```
Comment (ademarCardoso) [inline src/tasks/delete-datasources.js]:
```suggestion
const deleteDatasources = async (api, options) => {
```

File: src/tasks/delete-datasources.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      await api.deleteDatasource(datasource.id)
+    }
+
+  } catch (e) {
+    console.error(`${chalk.red('X')} An error ocurred in pull-components task when load components data`)
+    return Promise.reject(new Error(e))
+  }
+}
+
+module.exports = pullComponents
```
Comment (ademarCardoso) [inline src/tasks/delete-datasources.js]:
```suggestion
module.exports = deleteDatasources
```

File: src/tasks/delete-datasources.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      console.log(`${chalk.blue('-')} Datasources where name starts with ${byName}: ${formattedNames}`);
+    }
+
+    for (const datasource of datasources) {
+      console.log(`${chalk.blue('-')} Deleting ${datasource.name}`)
+      await api.deleteDatasource(datasource.id)
+    }
+
+  } catch (e) {
+    console.error(`${chalk.red('X')} An error ocurred in pull-components task when load components data`)
```
Comment (ademarCardoso) [inline src/tasks/delete-datasources.js]: Please update the message to math with the function shape.

File: src/cli.js [hunk 5]
```diff
@@ -505,6 +505,30 @@ program
...
+  try {
+    if (!api.isAuthorized()) {
+      await api.processLogin()
+    }
+
+    api.setSpaceId(spaceId)
+
+    await tasks.deleteDatasources(api, { byName, bySlug })
+  } catch (e) {
+    errorHandler(e, COMMANDS.PULL_COMPONENTS)
```
Comment (ademarCardoso) [inline src/cli.js]:
```suggestion
    errorHandler(e, COMMANDS.DELETE_DATASOURCES)
```

**Review by ademarCardoso — COMMENTED** [review]
Please take a look in my comments, i liked the feature 😄 

Thanks to the contribution 🚀

File: src/cli.js [hunk 6]
```diff
@@ -505,6 +505,30 @@ program
     }
   })
 
+  // delete-components
```
Comment (ademarCardoso) [inline src/cli.js]: ` // delete-datasources`

Comment (CaprinaeWT) [reply src/cli.js]: My bad 😅

**Review by ademarCardoso — APPROVED** [review]
